File: ruby/2.rb
# project euler 2: find all even fibonacci numbers below 4,000,000
t = Time.new.to_f
f1,f2,sum = 0,1,0
while f1<4000000
	f1,f2 = f2,f1+f2
	sum += f1 if f1.even?
end
puts "result: #{sum} - found in #{((Time.new.to_f - t)*1000).round 3} ms"

File: rust/2.rs
/*
 * Project Euler 2
 * Find the sum of all even Fibonacci numbers not exceeding 4,000,000
 */

fn main() {
    let mut sum = 0;
    let mut previous = 0;
    let mut current = 1;

    while current < 4000000 {
        if current % 2 == 0 {
            sum += current;
        }
        let temp = current;
        current = previous + current;
        previous = temp;
    }

    println!("Sum: {}", sum);
}
